modules/hpe_nimble_volume.py

```diff
@@ -0,0 +1,897 @@
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+# author alok ranjan (alok.ranjan2@hpe.com)
+
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+author:
+    - Alok Ranjan (@ranjanal)
+description: "Manage volume on a Nimble Storage group"
+module: hpe_nimble_volume
+options:
+    state:
+        description:
+        - "Choice for Volume operations"
+        choices:
+        - present
+        - absent
+        - create
+        - online
+        - offline
+        - restore
+        required: True
+        type: str
+    name:
+        required: True
+        type: str
+        description:
+        - "Name of the Source Volume."
+    size:
+        type: int
+        default: 100
+        description:
+        - "The size of the volume."
+    description:
+        required: False
+        type: str
+        default: null
+        description:
+        - Text description of volume.
+    perf_policy:
+        required: False
+        type: str
+        default: null
+        description:
+        - Name of the performance policy. After creating a volume, performance policy for the volume can only be
+          changed to another performance policy with same block size.
+    limit:
+        required: False
+        type: int
+        default: 100
+        description:
+        - Limit on the volume's mapped usage, expressed as a percentage of the volume's size
+    online:
+        required: False
+        type: bool
+        default: True
+        description:
+        - Online state of volume, available for host initiators to establish connections.
+    owned_by_group:
+        required: False
+        type: str
+        default: null
+        description:
+        - Name of group that currently owns the volume
+    multi_initiator:
+        required: False
+        type: bool
+        default: False
+        description:
+        - For iSCSI Volume Target, this flag indicates whether the volume and its snapshots can be accessed from multiple initiators at the same time.
+    iscsi_target_scope:
+        required: False
+        type: str
+        choices:
+            - volume
+            - group
+        default: 'volume'
+        description:
+        - This indicates whether volume is exported under iSCSI Group Target or iSCSI Volume Target. This attribute is only meaningful to iSCSI system
+    pool:
+        required: False
+        type: str
+        default: null
+        description:
+        - Name associated with the pool in the storage pool table.
```

Comment:
Pool should be capitalized if we capitalize Nimble objects.

Reply:
as a convention No variables in our module starts from Capital letter. It may be confusing while providing values in playbook if we have a mix of capital and non-capital variables.
as of now they just need to type in everything in small letters.
can we leave this as it is?

Reply:
yes, thanks for clarifying!